File: js/node/examples/event-builder.mjs
import {
  EventBuilder,
  Keys,
  Kind,
  KindStandard,
  SingleThreadPow,
} from "../dist/index.js";
import { runIfMain } from "./_main.mjs";

export function main() {
  const keys = Keys.generate();

  const builder = new EventBuilder(
    Kind.fromStd(KindStandard.TextNote),
    "Note from rust-nostr JavaScript bindings",
  );
  console.log(builder.finalize(keys).asJson());

  const customBuilder = new EventBuilder(new Kind(1234), "My custom content");
  console.log(`Event: ${customBuilder.finalize(keys).asJson()}`);

  const powEvent = customBuilder
    .finalizeUnsigned(keys.publicKey())
    .mine(new SingleThreadPow(), 8)
    .sign(keys);
  console.log(`POW event: ${powEvent.asJson()}`);

  console.log(
    `Unsigned event: ${customBuilder.finalizeUnsigned(keys.publicKey()).asJson()}`,
  );
}

await runIfMain(import.meta.url, main);

File: python/examples/event_builder.py
from nostr_sdk import *


def main():
    keys = Keys.generate()

    # Build a text note
    builder = EventBuilder(Kind.from_std(KindStandard.TEXT_NOTE), "Note from rust-nostr Python bindings")
    event = builder.finalize(keys)
    print(event.as_json())

    # Build a custom event
    kind = Kind(1234)
    content = "My custom content"
    builder = EventBuilder(kind, content)

    # Sign with generic signer
    event = builder.finalize(keys)
    print(f"Event: {event.as_json()}")

    # POW
    unsigned = builder.finalize_unsigned(keys.public_key())
    event = unsigned.mine(SingleThreadPow(), 8).sign(keys)
    print(f"POW event: {event.as_json()}")

    # Build unsigned event
    unsigned = builder.finalize_unsigned(keys.public_key())
    print(f"Unsigned event: {unsigned.as_json()}")


if __name__ == '__main__':
    main()
